user-impl/src/main/java/com/example/auction/user/impl/AuthCommand.java

```diff
@@ -0,0 +1,31 @@
+package com.example.auction.user.impl;
+
+import com.example.auction.user.api.Auth;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.lightbend.lagom.javadsl.persistence.PersistentEntity;
+import com.lightbend.lagom.serialization.Jsonable;
+import lombok.Value;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public interface AuthCommand extends Jsonable {
```

Comment:
Let's try to avoid `Auth` altogether. It is confusing because it could mean `Authentication` or `Authorization`.

In credential management it is very difficult to get naming correct.
In the past I found that a set of strings that a user sends to a server as an authentication could be named `Credentials`. So I'd have a `UsernamePasswordCredentials` or `Oauth2TokenCredentials` implementations.

Then, the data a user sends in a signup form are not yet credentials because we don't really know if they are valid: the username could be a duplicate, or the password too weak.

Etc...